crates/html-macro-test/src/lib.rs

```diff
@@ -332,3 +332,61 @@ fn single_branch_if_false_block() {
     }
     .test();
 }
+
+#[test]
+fn custom_component_props() {
+    struct Counter {
+        count: u8,
+    }
+
+    impl View for Counter {
+        fn render(&self) -> VirtualNode {
+            html! {
+                <span>Counter = {format!("{}", self.count)}</span>
+            }
+        }
+    }
+
+    let mut expected = VElement::new("div");
+    let mut child = VElement::new("span");
+    child.children = vec![VirtualNode::text("Counter = "), VirtualNode::text("1")];
+    expected.children = vec![child.into()];  
+
+    HtmlMacroTest {
+        desc: "Custom component with props",
+        generated: html! {
+          <div><Counter count={1}/></div>
+        },
+        expected: expected.into(),
+    }
+    .test();
+}
+
+#[test]
+fn custom_component_children() {
+    struct Child;
+
+    impl View for Child {
+        fn render(&self) -> VirtualNode {
+            html! {
+                <span></span>
```

Comment:
Hmm if this was `<br />` would you then end up with `<br>Hello World</br>` ?

That's the only thing preventing me from instant merging this right now since everything else looks GOOD.

I'm happy to take over this PR since it's been like half a year and this probably isnt on your radar haha. Just let me know!

Reply:
I suppose we could just panic if you try to insert a child into a self closing node ...